custom_components/moonraker/sensor.py

```diff
@@ -370,7 +370,7 @@ async def async_setup_optional_sensors(coordinator, entry, async_add_entities):
             desc = MoonrakerSensorDescription(
                 key=f"{split_obj[0]}_{split_obj[1]}",
                 status_key=obj,
-                name=split_obj[1].replace("_", " ").title(),
+                name=split_obj[1].replace("_", " ").title() + " Temperature",
```

Comment:
I'm adding the " Temperature" since the sensor can provide more than just temperature. For me, in Klipper I've got the sensor named "Enclosure", so the names of all the Home Assistant sensors become "Enclosure Temperature", "Enclosure Humidity", etc.

Without adding " Temperature" to the temp sensor, I either have the temperature labelled "Enclosure" and the humidity labelled "Enclosure Humidity" or with your example "MCU Temp" and "MCU Temp Humidity".

I suppose I could not add " Humidity" etc. to any of the names and just refer to the units, but that makes searching entities in HASS harder.

Reply:
I think if we want to go that path, you would need to remove the `_temp` from all of them

so it goes, `mcu_temperature`, `mcu_humidity`

Reply:
For my enclosure, there is no `_temp` as I did not include it in the name. My MCU doesn't report temp, but I guess `_temp` is automatically added by moonraker? If so, I can strip that out if present.

So basically instead of 
```
name=split_obj[1].replace("_", " ").title() + " Temperature",
```
we want
```
name=split_obj[1].removesuffix("_temp").replace("_", " ").title() + " Temperature",
```
right?

Reply:
actually, 

do we want to keep the format, so it introduce no change for existing users. 


example 
`name=split_obj[1].removesuffix("_temp").replace("_", " ").title() + "_temp",`

and for the others

`name=split_obj[1].removesuffix("_temp").replace("_", " ").title() + "_humidity",`
etc

Reply:
We could do that. I don't love the shorted "temp" instead of "temperature", but it seems a reasonable tradeoff to get the new functionality without damaging existing installs.

That said, I think the name field is the display name (hence the `replace("_", " ")`. I guess the sensor names are derived from that display name?